lib/reek/ast/sexp_extensions.rb

```diff
@@ -409,12 +403,49 @@ def text_name
         end
       end
 
+      # Utility methods for :module nodes.
+      module ModuleNode
+        include ModuleNodeBase
+        def name() self[1] end
+
+        def simple_name
+          name.is_a?(::Parser::AST::Node) ? name.simple_name : name
+        end
+      end
+
       # Utility methods for :class nodes.
       module ClassNode
         include ModuleNode
         def superclass() self[2] end
       end
 
+      # Utility methods for :casgn nodes.
+      module CasgnNode
+        include ModuleNodeBase
+
+        MODULE_DEFINERS = [:Class, :Struct]
+
+        def defines_module?
+          call = case value.type
+                 when :block
+                   value.call
+                 when :send
+                   value
+                 end
+          call &&
+            call.method_name == :new &&
```

Comment:
We have a guard clause like quite a few times in our code base. Could we move this into one speaking method called "initializer?" or something like that?